registry/handlers/manifests.go

```diff
@@ -481,15 +482,30 @@ func (imh *manifestHandler) applyResourcePolicy(manifest distribution.Manifest)
 
 }
 
-// DeleteManifest removes the manifest with the given digest from the registry.
+// DeleteManifest removes the manifest with the given digest or the tag with the given name from the registry.
 func (imh *manifestHandler) DeleteManifest(w http.ResponseWriter, r *http.Request) {
 	dcontext.GetLogger(imh).Debug("DeleteImageManifest")
 
-	if imh.Tag != "" {
+	if imh.App.isCache {
 		imh.Errors = append(imh.Errors, errcode.ErrorCodeUnsupported)
 		return
 	}
 
+	if imh.Tag != "" {
+		tagService := imh.Repository.Tags(imh.Context)
+		if err := tagService.Untag(imh.Context, imh.Tag); err != nil {
+			switch err.(type) {
+			case distribution.ErrTagUnknown, driver.PathNotFoundError:
```

Comment:
The tagStore will not raise the PathNotFoundError, see https://github.com/distribution/distribution/blob/main/registry/storage/tagstore.go#L118

For the 404 scenario, it should leverage the tagService.Lookup(), but it may introduce the performance problem.

Reply:
This is no longer the case. Have a look at https://github.com/distribution/distribution/pull/3427/files#r641027450